STMicroelectronics/STM32L4_L4+/app/az_ulib_dm/src/az_ulib_dm_blob_ustream_interface.c

```diff
@@ -284,26 +286,20 @@ AZ_NODISCARD az_result az_blob_create_ustream_from_blob(
   _az_PRECONDITION_NOT_NULL(ustream_data_cb);
   _az_PRECONDITION_NOT_NULL(blob_http_cb);
   _az_PRECONDITION_NOT_NULL(ip);
-  _az_PRECONDITION(strlen((char*)resource) > 0);
-  _az_PRECONDITION(strlen((char*)host) > 0);
+  _az_PRECONDITION_NOT_NULL(resource);
+  _az_PRECONDITION_NOT_NULL(host);
+  _az_PRECONDITION(resource[0] != 0);
```

Comment:
>_az_PRECONDITION(resource[0] != 0) [](http://example.com/codeflow?start=2&length=34)

what is the point of checking the first index like this?

Reply:
I'm replacing the strlen by 2 tests: first, test if the pointer is not NULL, and second, test if there is, at least, one non-null byte in the string. It is a better solution because if the pointer is NULL, strlen will crash instead of return 0, and strlen will run over the entire string but if the first char is non-null, the string is not empty.